### PR TITLE
Events

### PR DESCRIPTION
**Pull request by erika-rosaverde**
New module added that handle the events and the subscriptions. In particular, the new module contains the event_handler class that loads the subscriptions (if they are already stored), validates a new subscription before to store it, forwards the events to the subscribers. 
Core generate an unique ID that is assigned to the new objects when POST is called.
Backend_FS and tests has been updated to work with the new module.

### CODE REVIEW

**Review by mjaguil — APPROVED** [review]

File: sunfishcorelib/events/event_handler.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+               messageId = event["MessageId"]
+               origin = 'not_specified'
+
+               to_forward = []
+               if "OriginOfCondition" in event:
+                    origin = event["OriginOfCondition"]["@odata.id"]
+                    type = self.check_data_type(origin)
+                    if type in subscribtions["ResourceTypes"]:
+                         for id in subscribtions["ResourceTypes"][type]:
+                              to_forward.append(id)
```
Comment (christian-pinto) [inline sunfishcorelib/events/event_handler.py:249]: just use `extend()`
```python
to_forward.extend(subscribtions["ResourceTypes"][type])
```

same applies to all similar cases below

File: tests/test_sunfishcore_library.py [hunk 2]
```diff
@@ -2,90 +2,121 @@
...
+    # EVENTING and SUBSCRIPTIONS
+    def test_subscription(self):
+        path = os.path.join(self.conf['redfish_root'], self.conf["backend_conf"]["subscribers_root"])
+        assert self.core.create_object(path, tests_template.subscription)
+
+    @pytest.fixture(scope="session")
+    def httpserver_listen_address(self):
+        return ("localhost", 8080)
+    
+    def test_expected_request_handler(self, httpserver: HTTPServer):
```
Comment (christian-pinto) [inline tests/test_sunfishcore_library.py:119]: We should test having multiple subscriptions and handling an event targeting only one of them. To make sure the data structure we use for figuring out who to send to works fine.

File: tests/test_utils.py [hunk 3]
```diff
@@ -25,6 +25,12 @@ def check_delete(path):
...
 
+def get_id(root, collection):
+    list = os.listdir(os.path.join(os.getcwd(), root, collection))
+    for dir in list:
+        if dir != '.DS_Store' and dir != 'index.json':
+            return dir
+
 # def check_update(payload, redfish_root):
 #     path = os.path.join(get_resource_path(payload, redfish_root), 'index.json')
 #     modification_time = time.ctime(os.path.getctime(path))
```
Comment (christian-pinto) [inline tests/test_utils.py:36]: Remove commented code unless we have an explanation why we need it there as a comment.

**Review by christian-pinto — CHANGES_REQUESTED** [review]
Few things need to be improved but it is overall looking fine.

**Review by christian-pinto — APPROVED** [review]

**Comment by christian-pinto** [comment]
@mjaguil Could you please complete your review?

**Comment by mjaguil** [comment]
I’ll get it done, right now…


From: Christian Pinto ***@***.***>
Date: Friday, November 24, 2023 at 11:40 AM
To: OpenFabrics/sunfish_library_reference ***@***.***>
Cc: Aguilar, Michael James ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [OpenFabrics/sunfish_library_reference] Events (PR #7)

@mjaguil<https://github.com/mjaguil> Could you please complete your review?

**Review by mjaguil — APPROVED** [review]

**Pull request merged**